torchvision/csrc/io/decoder/stream.cpp

```diff
@@ -53,6 +56,46 @@ int Stream::openCodec(std::vector<DecoderMetadata>* metadata) {
     return ret;
   }
 
+  // multithreading heuristics
+  int max_threads = at::get_num_threads();
+  // first a safety check
+  if (num_threads > max_threads) {
+    num_threads = max_threads;
+  }
+
+  if (num_threads > 0) {
+    // if user defined, respect that
+    codecCtx_->thread_count = num_threads;
+    codecCtx_->active_thread_type = 1;
+  } else {
+    // otherwise set sensible defaults
+    // with the special case for the different MPEG4 codecs
+    // that don't have threading context functions
+    // TODO: potentially automate this using native c++ function lookups
+    if (strcmp(codecCtx_->codec->name, "mpeg4") == 0 &&
+        codecCtx_->codec_type == 0) {
+      if (codecCtx_->codec_tag == 1684633208) {
+        codecCtx_->thread_count = (8 <= max_threads) ? 8 : max_threads;
+        codecCtx_->thread_type = 1;
+      } else {
+        codecCtx_->thread_count = (2 <= max_threads) ? 2 : max_threads;
+        codecCtx_->thread_type = 2;
+      }
+    } else {
+      // otherwise default to multithreading
+      codecCtx_->thread_count = (8 <= max_threads) ? 8 : max_threads;
+      codecCtx_->active_thread_type = 1;
+    }
+  }
```

Comment:
Can we let the default behavior in C++ land to be the same as before, and have the custom operator wrapping in PyTorch to use multithreading by default?

The change in behavior for the C++-only for multi-threading codebase might have side-effects that I would rather avoid risking for now.